workspaces/marketplace/plugins/marketplace-backend/src/service/MarketplaceAggregationService.ts

```diff
@@ -0,0 +1,184 @@
+    baseQuery: Knex.QueryBuilder | null = null,
+  ): Promise<Knex.QueryBuilder> {
+    if (!this.client) {
+      throw new Error(
+        'Database client is not configured. Please check the MarketplaceCatalogClient configuration.',
+      );
+    }
+    const dbClient = this.client;
+    const query =
+      baseQuery ||
+      dbClient('final_entities as fe').whereNotNull('fe.final_entity');
+    aggregationsRequest.forEach((agg, index) => {
+      const currentAlias = `s${index + 1}`;
+
+      // Add LEFT JOIN for each aggregation
+      query.leftJoin(
+        `search as ${currentAlias}`,
+        'fe.entity_id',
+        `${currentAlias}.entity_id`,
+      );
+
+      const getLastValue = (input: string): string => {
+        if (!input.includes('.')) {
+          return input;
+        }
+
+        const parts = input.split('.');
+        return parts[parts.length - 1];
+      };
+
+      const fieldName = getLastValue(agg.name ?? agg.field);
+
+      // Add SELECT clause
+      if (agg.type === 'count') {
+        query.select(`${currentAlias}.value as ${fieldName}`);
+      }
+
+      // Apply aggregate functions
+      if (agg.type) {
+        if (agg.type === 'count') {
+          query.count('* as count');
+        } else if (['avg', 'sum'].includes(agg.type)) {
+          query
+            .select(
+              dbClient.raw(
+                `${agg.type.toLocaleUpperCase('en-US')}(CAST(${currentAlias}.value AS NUMERIC)) AS ${agg.type}`,
+              ),
+            )
+            .andWhereRaw(`${currentAlias}.value ~ ?`, ['^[0-9]+(\\.[0-9]+)?$']);
+        } else {
+          query.select(
+            dbClient.raw(
+              `${agg.type.toLocaleUpperCase('en-US')}(${currentAlias}.value) AS ${fieldName}`,
+            ),
+          );
+        }
+      }
+
+      // Apply WHERE filters
+      query.where(`${currentAlias}.key`, agg.field);
+      if (agg.value)
+        query.where(
+          `${currentAlias}.value`,
+          agg.value.toLocaleLowerCase('en-US'),
+        );
+
+      // Apply additional filters
+      if (agg.filter) {
+        Object.entries(agg.filter).forEach(([key, value], idx) => {
+          const filterAlias = `f${idx + 1}`;
+          query
+            .leftJoin(
+              `search as ${filterAlias}`,
+              'fe.entity_id',
+              `${filterAlias}.entity_id`,
+            )
+            .where(`${filterAlias}.key`, key)
+            .where(`${filterAlias}.value`, value.toLocaleLowerCase('en-US'));
+        });
+      }
+
+      // Add GROUP BY clause for count
+      if (agg.type === 'count') {
+        query.groupBy(`${currentAlias}.key`, `${currentAlias}.value`);
+      }
+
+      // Apply HAVING filters
+      if (agg.havingFilter) {
+        const { field, operator, value } = agg.havingFilter;
+        const rawField = field === 'count' ? dbClient.raw('COUNT(*)') : field;
+        query.having(rawField, operator, value);
+      }
```

Comment:
Do you think this should be an array and a foreach similar to the filter above?

Reply:
Yes, we can introduce support for multiple having clauses. Also I am thinking to add a new prop for logical operator (AND or OR) to both filters and having filters so it becomes more flexible.

Reply:
Yes, we can introduce support for multiple having clauses.